bluesky_queueserver/server/server.py

```diff
@@ -32,8 +32,20 @@
 async def startup_event():
     global zmq_to_manager
     global custom_code_module
+
+    # Read private key from the environment variable, then check if the CLI parameter exists
+    zmq_public_key = os.environ.get("QSERVER_ZMQ_PUBLIC_KEY", None)
+    zmq_public_key = zmq_public_key if zmq_public_key else None  # Case of ""
+    if zmq_public_key is not None:
+        try:
+            validate_zmq_key(zmq_public_key)
+        except Exception as ex:
+            raise ValueError("ZMQ public key is improperly formatted: %s", str(ex))
```

Comment:
This block of code is repeating in a few places. I think it'd make sense to put it in a single place and reuse it with a function.